test/miscellaneous_tests/api.jl

```diff
@@ -300,119 +300,6 @@ let
     testnetwork(rn, B, Z, Δ, lcs, 0, subrn, lcd)
 end
 
-### Tests Reversibility ###
-
-# Test function.
-function testreversibility(rn, B, rev, weak_rev)
-    @test isreversible(rn) == rev
-    subrn = subnetworks(rn)
-    @test isweaklyreversible(rn, subrn) == weak_rev
-end
-
-# Tests reversibility for networks with known reversibility.
-let
-    rn = @reaction_network begin
-        (k2, k1), A1 <--> A2 + A3
-        k3, A2 + A3 --> A4
-        k4, A4 --> A5
-        (k6, k5), A5 <--> 2A6
-        k7, 2A6 --> A4
-        k8, A4 + A5 --> A7
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        (k2, k1), A1 <--> A2 + A3
-        k3, A2 + A3 --> A4
-        k4, A4 --> A5
-        (k6, k5), A5 <--> 2A6
-        k7, A4 --> 2A6
-        (k9, k8), A4 + A5 <--> A7
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        k1, A --> B
-        k2, A --> C
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        k1, A --> B
-        k2, A --> C
-        k3, B + C --> 2A
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        (k2, k1), A <--> 2B
-        (k4, k3), A + C --> D
-        k5, D --> B + E
-        k6, B + E --> A + C
-    end
-    rev = false
-    weak_rev = true
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        (k2, k1), A + E <--> AE
-        k3, AE --> B + E
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        (k2, k1), A + E <--> AE
-        (k4, k3), AE <--> B + E
-    end
-    rev = true
-    weak_rev = true
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin (k2, k1), A + B <--> 2A end
-    rev = true
-    weak_rev = true
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        k1, A + B --> 3A
-        k2, 3A --> 2A + C
-        k3, 2A + C --> 2B
-        k4, 2B --> A + B
-    end
-    rev = false
-    weak_rev = true
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-let
-    rn = @reaction_network begin
-        (k2, k1), A + E <--> AE
-        (k4, k3), AE <--> B + E
-        k5, B --> 0
-        k6, 0 --> A
-    end
-    rev = false
-    weak_rev = false
-    testreversibility(rn, reactioncomplexes(rn)[2], rev, weak_rev)
-end
-
```

Comment:
Moved to network analysis tests. There were quite a few things in the API test file that I think maybe should be moved, but was honestly unsure. Only moved this and the conservation law stuff because those where the only ones I felt very certain about.